src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,116 @@
 public class StringManipulation implements StringManipulationInterface {
+    private String string;
 
-    @Override
     public String getString() {
-        return null;
+        return string;
     }
 
-    @Override
     public void setString(String string) {
+        this.string = string;
     }
 
-    @Override
+    /**
+     * Counts the number of non-empty words in the string.
+     * @return The count of words.
+     */
     public int count() {
-        return 0;
+        String[] words = string.split(" ");
+        int count = 0;
+
+        for (String word : words) {
+            if (!word.isEmpty()) {
+                count++;
+            }
+        }
+
+        return count;
     }
 
+    /**
+     * Removes the character at the nth position in the string.
+     * @param n The position of the character to be removed.
+     * @param maintainSpacing Determines whether to maintain spacing when removing characters.
+     * @return The string with the nth characters removed.
+     */
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
-    }
+        if (n <= 0) {
+            throw new IllegalArgumentException();
+        } else if (n > string.length()) {
+            throw new IndexOutOfBoundsException();
+        }
 
-    @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for (int i = 0; i < string.length(); i++) {
+            if ((i + 1) % n == 0) {
+                if (maintainSpacing) {
+                    stringBuilder.append(" ");
+                }
+            } else {
+                stringBuilder.append(string.charAt(i));
+            }
+        }
+
+        return stringBuilder.toString();
     }
 
+    /**
+     * Retrieves a substring of words from the specified startWord to the endWord.
+     * @param startWord The starting index of the substring.
+     * @param endWord The ending index of the substring.
+     * @return An array of substrings.
+     */
     @Override
-    public String restoreString(int[] indices) {
-        return null;
+    public String[] getSubStrings(int startWord, int endWord){
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord) {
+            throw new IllegalArgumentException();
+        }
+
+        String[] preFilteredResult = string.split(" ");
+        String[] substrings = new String[endWord - startWord + 1];
+        int wordCount = 1;
+        int substringsIndex = 0;
+        
+        for (String result : preFilteredResult) {
+            if (!result.isEmpty()) {
+                if (wordCount >= startWord && wordCount <= endWord) {
+                    substrings[substringsIndex] = result;
+                    substringsIndex++;
+                }
+                wordCount++;
+            }
+        }
```

Comment:
I'm certain your getSubStrings() works well and no need to change this time, but I'm just sharing an alternative method I used for a shorter code. I used copyOfRange() in the Array class, which copies the specified range of an array into another array.
For example, you could use the following code:

String[] substring = Arrays.copyOfRange(words, startWord-1, endWord);

(Note that you need to throw an IndexOutOfBoundsException() if the endWord is larger than the words length before using this statement.)